src/components/GeneralInfo/InfrastructureCard/InfrastructureCard.test.js

```diff
@@ -75,69 +76,74 @@ describe('InfrastructureCard', () => {
         entity: {},
       },
     });
-    const wrapper = mountWithRouter(<InfrastructureCard store={store} />);
-    expect(toJson(wrapper)).toMatchSnapshot();
+    const view = renderWithRouter(<InfrastructureCard store={store} />);
+    expect(view.asFragment()).toMatchSnapshot();
   });
 
   describe('api', () => {
-    it('should NOT call handleClick', () => {
-      const store = mockStore(initialState);
-      const onClick = jest.fn();
-      const wrapper = mountWithRouter(<InfrastructureCard store={store} />);
-      wrapper.find('dd a').first().simulate('click');
-      expect(onClick).not.toHaveBeenCalled();
-    });
-
-    it('should call handleClick on ipv4', () => {
+    it('should call handleClick on ipv4', async () => {
       const store = mockStore(initialState);
       const onClick = jest.fn();
       location.pathname = 'localhost:3000/example/ipv4';
       useParams.mockImplementation(() => ({ modalId: 'ipv4' }));
-      const wrapper = mountWithRouter(
+      renderWithRouter(
         <InfrastructureCard handleClick={onClick} store={store} />
       );
-      wrapper.find('dd a').first().simulate('click');
-      expect(onClick).toHaveBeenCalled();
+
+      await userEvent.click(
```

Comment:
Wouldn't it be better to wrap these `userEvent`s in `act`[0]?


[0] https://testing-library.com/docs/react-testing-library/api#act

Reply:
`act` is already called internally